// Copyright 2015 gRPC authors.

syntax = "proto3";

package grpc.testing;

option objc_class_prefix = "RMT";

message BoolValue {

  bool value = 1;
}

enum PayloadType {

  COMPRESSABLE = 0;
}

message Payload {

  PayloadType type = 1;

  bytes body = 2;
}

message EchoStatus {
  int32 code = 1;
  string message = 2;
}

message SimpleRequest {

  PayloadType response_type = 1;

  int32 response_size = 2;

  Payload payload = 3;

  bool fill_username = 4;

  bool fill_oauth_scope = 5;

  BoolValue response_compressed = 6;

  EchoStatus response_status = 7;

  BoolValue expect_compressed = 8;
}

message SimpleResponse {

  Payload payload = 1;

  string username = 2;

  string oauth_scope = 3;
}

message StreamingInputCallRequest {

  Payload payload = 1;

  BoolValue expect_compressed = 2;

}

message StreamingInputCallResponse {

  int32 aggregated_payload_size = 1;
}

message ResponseParameters {

  int32 size = 1;

  int32 interval_us = 2;

  BoolValue compressed = 3;
}

message StreamingOutputCallRequest {

  PayloadType response_type = 1;

  repeated ResponseParameters response_parameters = 2;

  Payload payload = 3;

  EchoStatus response_status = 7;
}

message StreamingOutputCallResponse {

  Payload payload = 1;
}

message ReconnectParams {
  int32 max_reconnect_backoff_ms = 1;
}

message ReconnectInfo {
  bool passed = 1;
  repeated int32 backoff_ms = 2;
}
